dhall-json/src/Dhall/JSON/Yaml.hs

```diff
@@ -57,7 +57,7 @@ parseDocuments :: Parser Bool
 parseDocuments =
   Options.Applicative.switch
             (   Options.Applicative.long "documents"
-            <>  Options.Applicative.help "If given a Dhall list, output a document for every element"
+            <>  Options.Applicative.help "If given a Dhall list, output a document for every element.  Each document, including the first one, will be preceded by \"---\", even if there is only one document"
```

Comment:
This still sounds like the option will only modify the behaviour when the input is a list, although it actually outputs non-list inputs as documents too. Sorry for being unclear about this aspect before!

But maybe this is not an important distinction, and the clarification isn't actually needed.